roles/build_project_repo/tasks/python_app_version.yml

```diff
@@ -0,0 +1,41 @@
+# Determine the version of the python app being deployed
+- name: Determine python app version
+  tags:
+    - python
+  block:
+    - name: Check if project has a version.py file
+      stat:
+        path: "{{ clone_root}}/{{ repo }}/{{ python_app }}/version.py"
+      register: python_version_file
+
+    - name: Get the version for the python package/app being deployed (app.version)
+      become: true
+      become_user: "{{ deploy_user }}"
+      shell:
+        cmd: "python3 -c 'from {{ python_app }} import version; print(version.__version__)'"
```

Comment:
One could also do `from . import version; ...` to look for version.py in the repo root; see comment above